wallet/db.c

```diff
@@ -659,15 +659,16 @@ static struct db_config *db_config_find(const char *driver_name)
 static struct db *db_open(const tal_t *ctx, char *filename)
 {
 	struct db *db;
-	const char *driver_name = "sqlite3";
 
 	db = tal(ctx, struct db);
 	db->filename = tal_strdup(db, filename);
 	list_head_init(&db->pending_statements);
+	if (!strstr(db->filename, "://"))
+		db_fatal("Could not extract driver name from \"%s\"", db->filename);
 
-	db->config = db_config_find(driver_name);
+	db->config = db_config_find(db->filename);
```

Comment:
This will do weird things if we ever introduce "sqlite-new://" or something.

I think we need to make a temp copy here and continue doing exact matches:

```
const char *sep;
...
sep = strstr(db->filename, "://");
if (!sep)
        db_fatal(...);
driver_name = tal_strndup(tmpctx, db->filename, sep - db->filename);
```